# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1346798608 14400
#      Tue Sep 04 18:43:28 2012 -0400
# Node ID 99db33cae82b6487a8ac44f760170a55d0ec8fe2
# Parent  27bdffcb41c5f3dbb7da597784e82c463c8cb16f
don't start the GUI if window system is disabled

* libinterp/octave.cc (octave_starting_gui): Check no_window_system.

diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -1078,17 +1078,17 @@ octave_execute_interpreter (void)
 }
 
 // Return int instead of bool because this function is declared
 // extern "C".
 
 int
 octave_starting_gui (void)
 {
-  if (! display_info::display_available ())
+  if (no_window_system || ! display_info::display_available ())
     return false;
 
   if (force_gui_option)
     return true;
 
   if (no_gui_option)
     return false;
 
